---
title: "Pathways analysis"
header-includes: #allows you to add in your own Latex packages
- \usepackage{float} #use the 'float' package
- \floatplacement{figure}{H} #make every figure with caption = h

output:
  pdf_document:
    fig_caption: true
    keep_tex: true
    latex_engine: pdflatex
  html_document:
    df_print: paged
urlcolor: blue
---

```{r setup.pathways, include=FALSE}
source("config/pathways_config.R")
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(cache = FALSE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(out.extra = '')
#knitr::opts_chunk$set(fig.pos = 'H')
knitr::opts_knit$set(root.dir = rootdir)
```

# Pathways analysis

I have taken up and down-regulated gene sets from the differential expression analysis and looked for pathway enrichment
using genesets from the [MSigDB](http://software.broadinstitute.org/gsea/msigdb/). Having looked at various sets there
is very little in the way of pathway enrichment for CPZ regulated genes (either in infection or not). For simplicity I
am just going to therefore show the pathways that are enriched for the salmonella infection conditions using the HALLMARK
geneset (version 5.1). This is the bet curated gene set of very well described biological pathways and therefore I think
is reasonable for showing that we see an expected response to Salmonella.

### HALLMARK gene sets


```{r pathways, fig.cap="Significantly enriched HALLMARK pathways in Salmonella infection (up-regulated)", fig.height=8, fig.width=6, echo=FALSE, message=FALSE}

source(pathways.helper)
library(ggplot2)
library(gridExtra)

pathway.table.sal <- read.csv("/gfs/work/nilott/proj004/analysis/rnaseq/enrichments2/sal.control.up/hallmarks/h.all.v5.1.symbols.foreground.results", header=T, stringsAsFactors=F, sep="\t")
p1 <- plotPathways(pathway.table.sal)
p1 <- p1 + ggtitle("Salmonella vs. Control") + coord_flip()
#ggsave("pathways_sal_vs_control.pdf")

pathway.table.cpz <- read.csv("/gfs/work/nilott/proj004/analysis/rnaseq/enrichments2/cpz.control.up/hallmarks/h.all.v5.1.symbols.foreground.results", header=T, stringsAsFactors=F, sep="\t")
p2 <- plotPathways(pathway.table.cpz)
p2 <- p2 + ggtitle("CPZ vs. Control") + coord_flip()

pathway.table.salcpz <- read.csv("/gfs/work/nilott/proj004/analysis/rnaseq/enrichments2/cpzSalmonella.salmonella.up/hallmarks/h.all.v5.1.symbols.foreground.results", header=T, stringsAsFactors=F, sep="\t")
p3 <- plotPathways(pathway.table.salcpz)
p3 <- p3 + ggtitle("CPZ + Salmonella vs. Control") + coord_flip()

lay <- matrix(c(1,1), c(2,2), c(3,3))
grid.arrange(grobs=list(p1, p2, p3), layout.matrix=lay)

```

### Genes in TNF pathway regualated by CPZ treamtment

Contrary to previous analysis it appears that there is a significant enrichment of TNFa signalling in CPZ treated macrophages. Here
we look at which genes these are and their pattern of regulation.

```{r tnf, fig.cap="Regulation of TNF pathway genes", fig.height=12, fig.width=12, echo=FALSE, message=FALSE}
source("/gfs/devel/nilott/NGSKit/R/pathways_helper.R")

genesets <- read.csv("/gfs/work/nilott/proj004/analysis/rnaseq/enrichments2/cpzSalmonella.salmonella.up/h.all.v5.1.symbols.tsv", header=F, stringsAsFactors = F, sep="\t")

tnf.genes <- getGenesetGenes(genesets, "HALLMARK_TNFA_SIGNALING_VIA_NFKB")

cpz.control.regulated.tnf <- intersect(as.character(cpz.control.up), tnf.genes)
salcpz.sal.regulated.tnf <- intersect(as.character(salcpz.sal.up), tnf.genes)

# collect grobs for figure
grob.list <- list()
for (i in cpz.control.regulated.tnf){
    goi <- i
    p <- plotGOI(rld2, goi=goi) + ggtitle("CPZ vs. Control")
    grob.list[[i]] <- p
    }

grid.arrange(grobs=grob.list, ncol=3, nrow=3)

# collect grobs for figure
grob.list <- list()
for (i in salcpz.sal.regulated.tnf){
    goi <- i
    p <- plotGOI(rld2, goi=goi) + ggtitle("Sal + CPZ vs. Sal")
    grob.list[[i]] <- p
    }

grid.arrange(grobs=grob.list, ncol=3, nrow=3)


```

